version/new_version.sh

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env bash
+
+# new_version.sh
+#
+# A helper script to update and generate version files for RStudio. 
+# It should be run from the root directory of the repository.
+#
+# Running the script
+# ------------------
+# ./new_version.sh <release flower> <calendar version> [base commit sha]
+#     i.e., ./new_version.sh "Cherry Blossom" 2023.03 [base commit sha]
+#
+# The base commit sha is optional. If it is not provided, the script will use the current commit hash.
+#   A base commit sha SHOULD be provided if a release branch for the calendar version already exists, 
+#   and should be the commit hash of the fork point for the release branch. 
+#   This is to ensure we "count" the work that was done on the release branch before starting builds.
```

Comment:
I've tried to do this. It's only straightforward if there haven't been multiple merges between the two branches... for instance we started work on `cherry-blossom` back with the commit that is a part of these changes. But trying to compute that commit from something like `git merge-base main cherry-blossom` doesn't actually give us that commit because our git history is not that clean.